### DISCUSSION TITLE
The provided value for the column is too long for the column's type. Column: desc

### DISCUSSION THREAD

**Discussion by CompeyDev** [prisma/prisma#14783] [Q&A] [answered]
I'm using JWT to sign tokens, and store them in a database using Prisma Client. JWT Tokens are 140 characters long. But when I attempt to store them, this is the error I get:
![image](https://user-images.githubusercontent.com/74418041/184356573-9498a6f0-1f14-4882-8e93-313396f7ca2b.png)

I tried increasing the varchar count to higher numbers, but the error still occurs.

Below is the scheme for the model:
```prisma
model User {
  id          Int      @id @default(autoincrement())
  sessionKey  String?  @unique
  sessionTime String?  @db.VarChar(255)
  token       String   @unique @db.VarChar(140)
  createdAt   DateTime @default(now())
  name        String   @db.VarChar(100)
}
```

**Comment by nurul3101** [comment]
Hey @CompeyDev 👋 
It looks like the value provided for column named `desc` is too long, but I can't see the field named `desc` in the User model. Is your PrismaClient in sync with your schema file?

**Comment by CompeyDev** [comment]
It is synced; a `desc` field doesn't exist in any models. But the only change I'd made was regarding tokens, so I'm assuming that's the reason of the error (changing the tokens from JWT fixes it?).

**Comment by CompeyDev** [comment]
Any ideas on how I can fix this permanently?

**Reply by janpio** [reply]
The error message indicates that there is a column `desc` in play somehow, so you will have to figure that out to be able to make any progress.

**Reply by CompeyDev** [reply]
That is extremely odd though, there's no desc columns anywhere in any of my prisma models. Maybe this could be a PlanetScale error where it is giving a default name of "desc" to a column it can't fetch? Because usually PlanetScale hosted databases don't provide proper column names with errors.

**Reply by janpio** [reply]
Puh, that is a good point. I have no idea really.

Can you create a minimal reproduction project of this problem so we can play with this ourselves? Something is off here and we should probably fix this.

**Reply by CompeyDev** [reply]
Actually, you can find the original repository at https://github.com/datalinkhq/datalink. It's a backend API written in Next.js.

**Reply by janpio** [reply]
What are the steps to get to this error? What do we have to do?

**Reply by CompeyDev** [reply]
Simply run the project with `pnpm dev` and make a post request to the `/api/create` endpoint with the JSON body:
```json
{
   "name": "someusername",
   "password": "somepasswordhere16253EW@"
}
```

The error should occur.

**Reply by pantharshit00** [reply]
Hello,

I checked the application code and the token that you are creating is of 151 characters in length while your `token` column is only of length `140`.

I got the correct column in the error when I tried to reproduce the problem:
```
The provided value for the column is too long for the column's type. Column: token
    at RequestHandler.handleRequestError (/Users/harshitpant/code/reproductions/issue-14783/datalink/node_modules/.pnpm/@prisma+client@4.2.1_prisma@4.2.1/node_modules/@prisma/client/runtime/index.js:28838:13)
    at RequestHandler.request (/Users/harshitpant/code/reproductions/issue-14783/datalink/node_modules/.pnpm/@prisma+client@4.2.1_prisma@4.2.1/node_modules/@prisma/client/runtime/index.js:28820:12)
    at async PrismaClient._request (/Users/harshitpant/code/reproductions/issue-14783/datalink/node_modules/.pnpm/@prisma+client@4.2.1_prisma@4.2.1/node_modules/@prisma/client/runtime/index.js:29753:16)
    at async give (webpack-internal:///(api)/./src/lib/setToken.ts:24:9)
    at async handler (webpack-internal:///(api)/./src/pages/api/create.ts:21:23)
    at async /Users/harshitpant/code/reproductions/issue-14783/datalink/node_modules/.pnpm/@sentry+nextjs@7.11.1_opuzalm7bqy6lx7uvhtk6ci2fu/node_modules/@sentry/nextjs/cjs/utils/withSentry.js:78:29
    at async Object.apiResolver (/Users/harshitpant/code/reproductions/issue-14783/datalink/node_modules/.pnpm/next@12.2.5_rlaikcoslzwwtqyk7brpdzej5y/node_modules/next/dist/server/api-utils/node.js:184:9)
    at async DevServer.runApi (/Users/harshitpant/code/reproductions/issue-14783/datalink/node_modules/.pnpm/next@12.2.5_rlaikcoslzwwtqyk7brpdzej5y/node_modules/next/dist/server/next-server.js:403:9)
    at async Object.fn (/Users/harshitpant/code/reproductions/issue-14783/datalink/node_modules/.pnpm/next@12.2.5_rlaikcoslzwwtqyk7brpdzej5y/node_modules/next/dist/server/base-server.js:493:37)
    at async Router.execute (/Users/harshitpant/code/reproductions/issue-14783/datalink/node_modules/.pnpm/next@12.2.5_rlaikcoslzwwtqyk7brpdzej5y/node_modules/next/dist/server/router.js:222:36) {
  code: 'P2000',
  clientVersion: '4.2.1',
  meta: { column_name: 'token' }
}
```

I changed `token       String   @unique @db.VarChar(140)` to `token       String   @unique @db.VarChar(202)` in the schema and it works as expected:
<img width="1488" alt="image" src="https://user-images.githubusercontent.com/22195362/186633918-da4606ca-0741-484c-800d-1b89c28220ba.png">

**Reply by CompeyDev** [reply]
> I checked the application code and the token that you are creating is of 151 characters in length while your token column is only of length 151.

However, when I tried increasing the column value, **this does not work at all**. Even after regenerating client multiple times.

**Reply by janpio** [reply]
How exactly did you do that? 
Did you also migrate/update your underlying database to match the changed schema, or only the generated Client?

**Reply by pantharshit00** [reply]
Can you also share exact DB version?

**Reply by CompeyDev** [reply]
Migrated, and it seems to work after reinstalling dependencies, thanks!

**Comment by CompeyDev** [comment, marked as answer]
Solved with: https://github.com/prisma/prisma/discussions/14783#discussioncomment-3396138

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
